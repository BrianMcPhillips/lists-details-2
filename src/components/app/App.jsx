import React from 'react';
import Container from '../../container/Container';
import { 
  BrowserRouter as Router,
  Route,
  Switch, 
} from 'react-router-dom';
import CharacterDetail from '../../container/CharacterDetail';
import Header from '../Header/Header';

export default function App() {
  return (
    <Router>
      <Switch>
        <Header/>
        <Route exact path="/" component={Container}/>
        <Route exact path="/detail/:id" component={CharacterDetail}/>
      </Switch>
    </Router>
  );
}
