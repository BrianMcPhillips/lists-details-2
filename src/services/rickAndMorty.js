export const getAllCharacters = () => {
  return fetch('https://rickandmortyapi.com/api/character/')
    .then(res => res.json())
    .then(json => json.results.map(character => ({
      name: character.name,
      status: character.status,
      species: character.species,
      image: character.image

    })));
};


export const getSingleCharacter = id => {
  return fetch(`https://rickandmortyapi.com/api/character/${id}`)
    .then(res => res.jsonS());
};

