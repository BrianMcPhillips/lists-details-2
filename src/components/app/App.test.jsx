import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';
import { getAllCharacters } from '../../services/rickAndMorty';

jest.mock('../../services/rickAndMorty.js');

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    getAllCharacters.mockResolvedValue([
      {
        name: 'test',
        status: 'test',
        species: 'test',
        image: 'https://www.placecage.com/200/200'
      },
      {
        name: 'test2',
        status: 'test2',
        species: 'test2',
        image: 'https://www.placecage.com/200/200'
      }
    ]);
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
