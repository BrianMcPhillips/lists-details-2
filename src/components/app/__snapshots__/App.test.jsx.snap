// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <h1>
    Rick and Morty Characters
  </h1>
  <a
    href="/"
  >
    <span>
      Home
    </span>
  </a>
</DocumentFragment>
`;
